cli/commands/project_create.go

```diff
@@ -41,26 +49,57 @@ func (cmd *ProjectCreate) Create(ctx *cli.Context) error {
 		image = "outrigger/generator"
 	}
 
+	argsMessage := " with no arguments"
+	if ctx.Args().Present() {
+		argsMessage = fmt.Sprintf(" with arguments: %s", strings.Join(ctx.Args(), " "))
+	}
+
+	if cmd.machine.IsRunning() {
+		cmd.out.Verbose.Printf("Executing container %s%s", image, argsMessage)
+		cmd.RunGenerator(ctx, cmd.machine, image)
+	} else {
+		cmd.out.Error.Fatalf("Machine '%s' is not running.", cmd.machine.Name)
+	}
+
+	return nil
+}
+
+func (cmd *ProjectCreate) RunGenerator(ctx *cli.Context, machine Machine, image string) error {
+	machine.SetEnv()
+
+	// The check for whether the image is older than 30 days is not currently used.
+	_, seconds, err := util.ImageOlderThan(image, 86400*30)
+	if err == nil {
+		cmd.out.Verbose.Printf("Local copy of the image '%s' was originally published %0.2f days ago.", image, seconds/86400)
+	}
+
+	// If there was an error it implies no previous instance of the image is available
+	// or that docker operations failed and things will likely go wrong anyway.
+	if err == nil && !ctx.Bool("no-update") {
+		cmd.out.Verbose.Printf("Attempting to update %s", image)
+		if err := util.StreamCommand(exec.Command("docker", "pull", image)); err != nil {
+			cmd.out.Verbose.Println("Failed to update generator image. Will use local cache if available.")
+		}
+	} else if err == nil && ctx.Bool("no-update") {
+		cmd.out.Verbose.Printf("Automatic generator image update suppressed by --no-update option.")
+	}
+
 	cwd, err := os.Getwd()
 	if err != nil {
 		cmd.out.Error.Printf("Couldn't determine current working directory: %s", err)
 		os.Exit(1)
 	}
 
-	argsMessage := " with no arguments"
-	if ctx.Args().Present() {
-		argsMessage = fmt.Sprintf(" with arguments: %s", strings.Join(ctx.Args(), " "))
-	}
-	cmd.out.Verbose.Printf("Executing container %s%s", image, argsMessage)
-
-	// keep passed in args as distinct elements or they will be treated as
-	// a single argument containing spaces when the container gets them
+	// Keep passed in args as distinct elements or they will be treated as
+	// a single argument containing spaces when the container gets them.
 	args := []string{
 		"container",
 		"run",
 		"--rm",
 		"-it",
 		"-v", fmt.Sprintf("%s:/generated", cwd),
+		"--name",
+		generator_container_name,
```

Comment:
Since we name the Dashboard container. Since this might actually be done concurrently I'll remove that.